app/controllers/services_controller.rb

```diff
@@ -0,0 +1,60 @@
+class ServicesController < ApplicationController
+  def index
+    @services = Service.all
+  end
+
+  def create
+    if user_signed_in?
```

Comment:
a DRY way of doing this will be by create a private method that checks authentication, and then using a filter at the top of the controller. Please read http://guides.rubyonrails.org/action_controller_overview.html#filters